x-pack/test/security_solution_cypress/cypress/e2e/entity_analytics/new_entity_flyout.cy.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  expandFirstAlertHostFlyout,
+  expandFirstAlertUserFlyout,
+} from '../../tasks/asset_criticality/common';
+import { login } from '../../tasks/login';
+import { visitWithTimeRange } from '../../tasks/navigation';
+import { ALERTS_URL } from '../../urls/navigation';
+import { enableRiskEngine } from '../../tasks/entity_analytics';
+import { deleteRiskEngineConfiguration } from '../../tasks/api_calls/risk_engine';
+import { USER_PANEL_HEADER } from '../../screens/hosts/flyout_user_panel';
+import { waitForAlerts } from '../../tasks/alerts';
+import { HOST_PANEL_HEADER } from '../../screens/hosts/flyout_host_panel';
+import { RISK_INPUT_PANEL_HEADER } from '../../screens/flyout_risk_panel';
+import { expandRiskInputsFlyoutPanel } from '../../tasks/risk_scores/risk_inputs_flyout_panel';
+
+const USER_NAME = 'user1';
+const SIEM_KIBANA_HOST_NAME = 'Host-fwarau82er';
+
+describe(
+  'Entity Flyout',
+  {
+    tags: ['@ess', '@serverless'],
+    env: {
+      ftrConfig: {
+        kbnServerArgs: [
+          `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+            'newUserDetailsFlyout',
+            'newHostDetailsFlyout',
+          ])}`,
+        ],
+      },
+    },
+  },
+  () => {
+    before(() => {
+      login();
+      enableRiskEngine();
```

Comment:
I think that Glo mentioned that the only thing we should do in the _before_ and _after_ hooks is loading archives, anything else (like `enableRiskEngine` or `deleteRiskEngineConfiguration` calls) should go to _beforeEach_ and _afterEach_, is this correct @MadameSheema?

Reply:
correct @semd

Reply:
Unfortunately, we cannot load the archived data without enabling the risk engine first. 

The only thing I can do is move everything to `beforeEach` and `afterEach,` which will slow down the test execution and could create flakiness. The problem here is that we don't have a way to delete every resource that is created when we enable the engine (we should have that). So when we enable the engine for the second time, it keeps the previously created data, and that could impact the test.

Could you please share the documentation for this `before` and `after` rule? In this case, how can we execute some logic only once inside Cypress tests?

Reply:
@MadameSheema @semd I decided to change the test approach. Now, instead of enabling the risk engine, I am mocking the risk engine status API. That will make the test more reliable. I also fixed some archiver bugs that prevented the test from running twice.

What do you think?

https://github.com/elastic/kibana/pull/174232/commits/edd34473ae47a3655d94429d1326053b1293ad3d#diff-145ec61fabe13c482eb0a6a759f06b3455b61578d4b32135f3c1363bccbae2a9L62-L69

Reply:
> Could you please share the documentation for this before and after rule? 

Hey @machadoum!! I'm working on more extensive documentation on automation best practices, but you can find that statement [here](https://github.com/elastic/kibana/issues/173327) and [here](https://github.com/elastic/kibana/issues/174247). 

> In this case, how can we execute some logic only once inside Cypress tests?

Can you be more specific about cases where you want to execute a logic just once?

> @MadameSheema @semd I decided to change the test approach. Now, instead of enabling the risk engine, I am mocking the risk engine status API. That will make the test more reliable. I also fixed some archiver bugs that prevented the test from running twice.

What do you think?

As the scope of the test is not testing the process of enabling the risk engine, I believe is a great idea towards having more reliable and stable tests :)

Reply:
@machadoum @MadameSheema 
That's a good temporary solution, it will prevent flakiness. However, I think it would be better if we were able to test the real API, to make sure we are testing the UI -> API integration property. 

Nevertheless, from what I know, and correct me if I am wrong, installing the risk engine is not a quick task, it may take some minutes. So, installing it in the beforeEach and removing it in the afterEach test case is not the best approach. 

What do you think if we unblock this PR and merge it with the API mock, and also create a separate issue to implement a helper/util library to install the risk engine in the _before_ hook and clean at the end in the _after_ hook?

Reply:
It works for me. 👍 

I brought to the team's attention the need for an API that completely deletes all risk engine assets. If we have this API we could call it once inside the `after` block.